packages/react-components/src/components/VideoGallery.test.tsx

```diff
@@ -678,6 +678,92 @@ describe('VideoGallery with vertical overflow gallery tests', () => {
+      overflowGalleryPosition: 'verticalRight',
+      maxRemoteVideoStreams: 2
+    };
+    const { rerender, container } = render(<VideoGallery {...videoGalleryProps} />);
+    rerender(<VideoGallery {...videoGalleryProps} dominantSpeakers={['2', '3']} />);
+
+    const gridTiles = getGridTiles(container);
+    expect(gridTiles.length).toBe(0);
+    const verticalGalleryTiles = getTiles(getVerticalGallery(container));
+    expect(verticalGalleryTiles.length).toBe(4);
+    expect(getDisplayName(verticalGalleryTiles[0])).toBe('Remote Participant 0');
+    expect(tileIsVideo(verticalGalleryTiles[0])).toBe(false);
+    expect(getDisplayName(verticalGalleryTiles[1])).toBe('Remote Participant 1');
+    expect(tileIsVideo(verticalGalleryTiles[1])).toBe(false);
+    expect(getDisplayName(verticalGalleryTiles[2])).toBe('Remote Participant 2');
+    expect(tileIsVideo(verticalGalleryTiles[2])).toBe(true);
+    expect(getDisplayName(verticalGalleryTiles[3])).toBe('Remote Participant 3');
+    expect(tileIsVideo(verticalGalleryTiles[3])).toBe(true);
+  });
+
+  test('should render video of dominant speaker when some participant is spotlighted', () => {
+    const localParticipant = createLocalParticipant({
+      videoStream: { isAvailable: true, renderElement: createVideoDivElement() }
+    });
+    // 10 remote participants. First 5 with their video on.
+    const remoteParticipants = [...Array(10).keys()].map((i) => {
+      return createRemoteParticipant({
+        userId: `${i}`,
+        displayName: `Remote Participant ${i}`,
+        videoStream: i < 5 ? { isAvailable: true, renderElement: createVideoDivElement() } : undefined
+      });
+    });
+
+    const videoGalleryProps: VideoGalleryProps = {
+      layout: 'floatingLocalVideo',
+      localParticipant,
+      remoteParticipants,
+      overflowGalleryPosition: 'verticalRight',
+      maxRemoteVideoStreams: 2,
+      spotlightedParticipants: ['9']
+    };
+    const { rerender, container } = render(<VideoGallery {...videoGalleryProps} />);
+    rerender(<VideoGallery {...videoGalleryProps} dominantSpeakers={['2', '3']} />);
+
+    const gridTiles = getGridTiles(container);
+    expect(gridTiles.length).toBe(1);
+    expect(getDisplayName(gridTiles[0])).toBe('Remote Participant 9');
+    const verticalGalleryTiles = getTiles(getVerticalGallery(container));
+    expect(verticalGalleryTiles.length).toBe(4);
+    expect(getDisplayName(verticalGalleryTiles[0])).toBe('Remote Participant 0');
+    expect(tileIsVideo(verticalGalleryTiles[0])).toBe(false);
+    expect(getDisplayName(verticalGalleryTiles[1])).toBe('Remote Participant 1');
+    expect(tileIsVideo(verticalGalleryTiles[1])).toBe(false);
+    expect(getDisplayName(verticalGalleryTiles[2])).toBe('Remote Participant 2');
+    expect(tileIsVideo(verticalGalleryTiles[2])).toBe(true);
+    expect(getDisplayName(verticalGalleryTiles[3])).toBe('Remote Participant 3');
+    expect(tileIsVideo(verticalGalleryTiles[3])).toBe(true);
+  });
```

Comment:
Add test for spotlighted participant is showing their stream and also a spotlighted participant without their stream showing

Reply:
Also for the other layouts

Reply:
... in another PR